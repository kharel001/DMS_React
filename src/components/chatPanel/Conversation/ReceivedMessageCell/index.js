import React from 'react';

const ReceivedMessageCell = ({conversation, user}) => {
  return (
    <div className="d-flex flex-nowrap chat-item">

      <img className="rounded-circle avatar size-40 align-self-end" src={user.thumb}
           data-test="imageComp" alt=""/>

      <div className="bubble">
        <div className="message" data-test="messageComp">{conversation.message}</div>
        <div className="time text-muted text-right mt-2" data-test="sentAtComp">{conversation.sentAt}</div>
      </div>

    </div>
  )
};

export default ReceivedMessageCell;