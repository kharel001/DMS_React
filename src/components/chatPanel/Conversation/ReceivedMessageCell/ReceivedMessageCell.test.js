import React from 'react';
import ReceivedMessageCell from './index';
import {
    configure,
    shallow,
} from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
// JestHook.mock('expo-font');
configure({
    adapter: new EnzymeAdapter
})

const dummy_data = {
    conversation:{
        message:"hello, how are you doing?",
        sentAt:"Wed 4:30"},
         user:{
             thumb:"thumbnail1"
         }
}
const setup = (props = {}) => {
    return shallow( < ReceivedMessageCell
     {...props}  />
    )
}

const findByTestAttr=(wrapper, val)=>{
    return wrapper.find(`[data-test='${val}']`)
}

{/* it("Custom Scrollable render", () => {
        let wrapper = setup(dummy_data);
        // expect(wrapper).toMatchSnapshot();
}) */}
it("Snapshot ReceivedMessageCell", () => {
    let wrapper = setup(dummy_data);
    expect(wrapper).toMatchSnapshot();

})



it("Components ReceivedMessageCell", () => {
    let wrapper = setup(dummy_data);
    let messageComp = findByTestAttr(wrapper, "messageComp");
    let sentAtComp = findByTestAttr(wrapper, "sentAtComp");
    let imageComp = findByTestAttr(wrapper, "imageComp");
   
    expect(sentAtComp.text()).toEqual("Wed 4:30");
    expect(messageComp.text()).toEqual("hello, how are you doing?");
    expect(imageComp.prop('src')).toEqual("thumbnail1")
})