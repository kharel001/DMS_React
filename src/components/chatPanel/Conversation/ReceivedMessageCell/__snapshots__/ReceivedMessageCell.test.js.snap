// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot ReceivedMessageCell 1`] = `ShallowWrapper {}`;
